compute/quickstart/compute_quickstart_sample_test.go

```diff
@@ -0,0 +1,180 @@
+// Copyright 2022 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+package main
+
+import (
+	"fmt"
+	"net/http"
+	"net/http/httptest"
+	"reflect"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"go.mongodb.org/mongo-driver/bson"
+	"go.mongodb.org/mongo-driver/mongo"
+	"go.mongodb.org/mongo-driver/mongo/integration/mtest"
+)
+
+var mt *mtest.T
+
+type mockManager struct{}
+
+func (m mockManager) load(t []trainer) error {
+	return nil
+}
+
+func (m mockManager) setCollection(c *mongo.Collection) {
+}
+
+func (m mockManager) list() ([]*trainer, error) {
+	trainers := []*trainer{
+		{"Ash", 10, "Pallet Town"},
+		{"Misty", 10, "Cerulean City"},
+		{"Brock", 15, "Pewter City"},
+	}
+
+	return trainers, nil
+}
+
+func newMockManager() mockManager {
+	m := mockManager{}
+	return m
+}
+
+func newTestManager() mongoManager {
+	m := &trainerManager{}
+	return m
+}
+
+// Testing mongo using mocks is explained here:
+// https://medium.com/@victor.neuret/mocking-the-official-mongo-golang-driver-5aad5b226a78
+func TestLoad(t *testing.T) {
+	mt := mtest.New(t, mtest.NewOptions().ClientType(mtest.Mock))
+	defer mt.Close()
+
+	tm := newTestManager()
+
+	mt.Run("success", func(mt *mtest.T) {
+		tm.setCollection(mt.Coll)
+		trainers := []trainer{
+			{
+				Name: "Ash",
+				Age:  10,
+				City: "Pallet Town",
+			},
+			{
+				Name: "Misty",
+				Age:  10,
+				City: "Cerulean City",
+			},
+			{
+				Name: "Brock",
+				Age:  15,
+				City: "Pewter City",
+			},
+		}
+		mt.AddMockResponses(mtest.CreateSuccessResponse())
+
+		err := tm.load(trainers)
+		assert.Nil(t, err)
+	})
+
+	mt.Run("err", func(mt *mtest.T) {
+		want := "error inserting records to mongo: must provide at least one element in input slice"
+		tm.setCollection(mt.Coll)
+		trainers := []trainer{}
+		mt.AddMockResponses(mtest.CreateSuccessResponse())
+
+		err := tm.load(trainers)
+		if !reflect.DeepEqual(want, err.Error()) {
```

Comment:
Are you looking for a substring match here?

Reply:
yes. I'm just used to using !reflect.DeepEqual(want, [got])) patern everywhere.  Cleaned it up/